### PR TITLE
0647 - HoloFuel - Decimal value rounding in numpad & Request Fees

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

File: src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js [hunk 1]
```diff
@@ -139,12 +141,12 @@ export default function CreateOfferRequest ({ history: { push } }) {
       <h4 styleName='amount-label'>
         {title}
       </h4>
-      <div styleName='amount'>
+      <div styleName='amount' onClick={() => setNumpadVisible(true)}>
```
Comment (JettTech) [inline src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js:144]: Nice addition.  Did you met with C to determine whether this should be accompanied with a icon to signal this functionality?

**Review by JettTech — CHANGES_REQUESTED** [review]
Almost there! :+1:  

At [line #22 in AmountInput.js](https://github.com/Holo-Host/hp-admin/blob/29577d00d02e9554d6fdf05518b26428d28336a3/src/holofuel/pages/CreateOfferRequest/AmountInput/AmountInput.js#L22), we need to handle the condition wherein we only have one digit in the InputValue string, and a user selects the `removeDigit` function.  (Otherwise, when processing the new InputValue inside the `presentHolofuelString`, the split method will fail and result in an error.) 

Example Implementation:
```
const isValidinputValueLength = inputValue.length > 1
const removeDigit = () => isValidinputValueLength ? setInputValue(String(inputValue).slice(0, -1)) :  setInputValue(String(0))
```

**Comment by JettTech** [comment]
Also, I noticed a small CSS bug.  It looks like the `.amount-input-amount ` class inside AmountInput.module.css needs to explicitly set its width to 100%.  Otherwise, the input box stretches well beyond the width of the screen and causes alignment and odd interaction/behavior while in a mobile view.

File: src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js (hunk 1 above)
Comment (robbiecarlton) [reply src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js:144]: no this is more like an easter egg. It just seemed like the right thing to do. There could definitely be some kind of design element to point people to this feature.

File: src/holofuel/pages/CreateOfferRequest/AmountInput/AmountInput.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export default function AmountInput ({ amount, setAmount, chooseSend, chooseRequ
   const [inputValue, setInputValueRaw] = useState(String(amount))
 
   const setInputValue = value => {
-    const cleanValue = value.replace(/[^0-9.]/g, '') || 0 // strips non numerical characters
+    const cleanValue = value.replace(/[^0-9.]/g, '') || '0' // strips non numerical characters
```
Comment (JettTech) [inline src/holofuel/pages/CreateOfferRequest/AmountInput/AmountInput.js:11]: :clap:

File: src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.test.js [hunk 3]
```diff
@@ -446,4 +446,22 @@ describe('AmountInput', () => {
     fireEvent.click(getByText('Send'))
     expect(getByText(`${presentHolofuelAmount('1.034')} TF`)).toBeInTheDocument()
   })
+
```
Comment (JettTech) [inline src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.test.js:449]: Good job on adding the extra test! :+1:

**Review by JettTech — APPROVED** [review]
LGTM!  Nice catch on the string type. :shipit:

**Pull request merged**